src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -689,7 +689,7 @@ private BoundStatement BindDeclarationStatementParts(LocalDeclarationStatementSy
         /// <returns>The <see cref="MethodSymbol"/> of the Dispose method if one is found, otherwise null.</returns>
         internal MethodSymbol TryFindDisposePatternMethod(BoundExpression expr, SyntaxNode syntaxNode, bool hasAwait, DiagnosticBag diagnostics)
         {
-            Debug.Assert(!(expr is null));
+            Debug.Assert(expr is object);
             Debug.Assert(!(expr.Type is null));
```

Comment:
Seems like this one should change too, am I missing something?

Reply:
Indeed.  I missed that case :)